### PR TITLE
Restore test with deleting source pvc

### PR DESCRIPTION
**Pull request by ShellyKa13**
Signed-off-by: Shelly Kagan <skagan@redhat.com>

**What this PR does / why we need it**:
This adds a functional test for verifying the fix for the following bug: 
https://bugzilla.redhat.com/show_bug.cgi?id=2104479 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/storage/restore.go [hunk 1]
```diff
@@ -1646,11 +1654,16 @@ var _ = SIGDescribe("VirtualMachineRestore Tests", func() {
...
 					checkCloneAnnotations(vm, true)
+					if deleteSourcePVC {
+						libstorage.DeleteDataVolume(&sourceDV)
+					}
 					doRestore("", console.LoginToCirros, offlineSnaphot, getTargetVMName(restoreToNewVM, newVmName))
 					checkCloneAnnotations(getTargetVM(restoreToNewVM), false)
 				},
-					Entry("to the same VM", false),
-					Entry("to a new VM", true),
+					Entry("to the same VM", false, false),
```
Comment (alicefr) [inline tests/storage/restore.go:1663]: I'd add the full description here, you are simply describing only one of the 2 booleans. We shouldn't have 2 entries with the same description

Comment (ShellyKa13) [reply tests/storage/restore.go:1663]: right I did it on the first 2 tests and forgot to update this 2 too :) thanks

**Pull request merged**